### ISSUE TITLE
Mapping for multi-occurrence elements doesn't trigger auto-generated mappings

### ISSUE THREAD

**Issue by GoogleCodeExporter** [vadbch/orika#109]
```
Registering mappers occasionally produces undetermined results due to 
non-ordered ConcurrentHashMap for mapperRegistry.

Mappers should be processed in a predictable manner by using an ordered hash 
map implementation.


```

Original issue reported on code.google.com by `matt.deb...@gmail.com` on 15 Jun 2013 at 6:03

**Comment by GoogleCodeExporter** [comment]
```
See attached test case
```

Original comment by `matt.deb...@gmail.com` on 16 Jun 2013 at 12:55

Attachments:
- [Issue109TestCase.java](https://storage.googleapis.com/google-code-attachments/orika/issue-109/comment-1/Issue109TestCase.java)

**Comment by GoogleCodeExporter** [comment]
Original comment by `matt.deb...@gmail.com` on 16 Jun 2013 at 12:59
- Changed title: **Mapping for multi-occurrence elements**

**Comment by GoogleCodeExporter** [comment]
Original comment by `matt.deb...@gmail.com` on 16 Jun 2013 at 1:04
- Changed title: **Mapping for multi-occurrence elements doesn't trigger auto-generated mappings**

**Comment by GoogleCodeExporter** [comment]
Original comment by `matt.deb...@gmail.com` on 27 Jun 2013 at 2:24
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Fixed  in 1.4.3 release
```

Original comment by `matt.deb...@gmail.com` on 27 Jun 2013 at 2:25
